dockstore-common/src/main/java/io/dockstore/common/yaml/DockstoreYamlHelper.java

```diff
@@ -209,27 +211,18 @@ private static <T> T readContent(final String content, final Constructor constru
             safeYaml.load(content);
             Representer representer = new Representer();
             representer.getPropertyUtils().setSkipMissingProperties(skipUnknownProperties);
-            final Yaml yaml = new Yaml(constructor, representer);
+            DumperOptions dumperOptions = new DumperOptions();
+            LoaderOptions loaderOptions = new LoaderOptions();
+            loaderOptions.setAllowDuplicateKeys(false);
+            final Yaml yaml = new Yaml(constructor, representer, dumperOptions, loaderOptions);
             return yaml.load(content);
         } catch (Exception e) {
-            final String exceptionMsg = createExceptionMessage(e);
+            final String exceptionMsg = e.getMessage();
             LOG.error(ERROR_READING_DOCKSTORE_YML + exceptionMsg, e);
             throw new DockstoreYamlException(exceptionMsg);
         }
     }
 
-    private static String createExceptionMessage(Exception e) {
-        // For yaml ConstructorException, use `getProblem()`, which returns the line number, cause, and snippet.
```

Comment:
What was the explanation for `getProblem()` no longer being useful?

Reply:
We fixed the UI so that pre-formatted error messages weren't getting mangled when displayed, so now, we might as well display the whole error message, which includes the result of `getProblem()`.